### PR TITLE
[Term Entry] Python Plotly-graph_objects .Mesh3d()

### PR DESCRIPTION
**Pull request by FelixOmollo**
### Description

-Added the Plotly-graph_objects .Mesh3d() module for Python.

### Issue Solved

-Closes issue #5178 

### Type of Change

- Adding a new entry

### Checklist

- [x ] All writings are my own.
- [ x] My entry follows the Codecademy Docs style guide.
- [ x] My changes generate no new warnings.
- [ x] I have performed a self-review of my own writing and code.
- [ x] I have checked my entry and corrected any misspellings.
- [x ] I have made corresponding changes to the documentation if needed.
- [ x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x ] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by mamtawardhani** [comment]
> [CLA assistant check](https://cla-assistant.io/Codecademy/docs?pullRequest=5201) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=5201) before we can accept your contribution.
> 
> **Your Name** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=5201) it.

@FelixOmollo please sign the CLA, so that we can start with the review for this PR, thank you 😀

**Comment by FelixOmollo** [comment]
Hello, signed the CLA. Looking forward to your review. Thank you.

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+- 'https://www.codecademy.com/learn/paths/computer-science'
+
+- 'https://www.codecademy.com/learn/learn-python-3'
+
+- 'https://plotly.com/python/getting-started/'
+
+- 'https://plotly.github.io/plotly.py-docs/generated/plotly.graph_objects.Mesh3d.html'
+
+
+---
```
Comment (mamtawardhani) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:25]:
```suggestion
---
Title: '.Mesh3d()'
Description: 'Creates 3D triangular mesh plots for visualizing 3D surfaces or volumetric data.'
Subjects:
- 'Data Science'
- 'Data Visualization'
Tags:
- 'Charts'
- 'Data'
- 'Plotly'
CatalogContent:
- 'learn-python-3'
- 'paths/computer-science'
---
```

Comment (mamtawardhani) [reply content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:25]: The tags, subjects and catalog are needed to be added from here: [tags](https://github.com/Codecademy/docs/blob/main/documentation/tags.md), [subjects](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md) and c[atalog content](https://github.com/Codecademy/docs/blob/main/documentation/catalog-content.md)

Also, a short description of the term has to be added, I've added one, you can change it, that is just for reference.

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+The **`.Mesh3d()`** is a function in Plotly's graph-objects module used to create 3D mesh plots. The surface is constructed from a set of vertices in 3D space `(x,y,z)` and triangles that are defined by three points(vertices). These triangles are specified using index arrays `(i, j, k)`, where each index refers to a vertex.
```
Comment (mamtawardhani) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:27]:
```suggestion
The **`.Mesh3d()`** function in Plotly's graph-objects module is used to create 3D mesh plots. The surface is constructed from a set of vertices in 3D space `(x, y, z)` and triangles defined by three vertices. These triangles are specified using index arrays `(i, j, k)`, where each index refers to a vertex.
```

Comment (mamtawardhani) [reply content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:27]: reframed for better readability

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+## Syntax
+
+```pseudo
+
+plotly.graph_objects.Mesh3d(x=None, y=None, z=None, i=None, j=None, k=None, color=None, opacity=None, ...)
+
+
```
Comment (mamtawardhani) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:35]:
```suggestion
plotly.graph_objects.Mesh3d(x=None, y=None, z=None, i=None, j=None, k=None, color=None, opacity=None, ...)
```

Comment (mamtawardhani) [reply content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:35]: removed the empty spacing

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+-  `i, j, k`: These specify how the vertices should be connected to form triangles. Each triangle is defined by a triplet of indices `(i[j], j[j], k[j]),` where each index refers to a vertex in the `x, y, z` arrays.
+
+-  `color`: Specifies the color of the mesh.
+
+-  `opacity`: Controls the transparency of the mesh (ranges from 0 to 1, where 1 is fully opaque and 0 is fully transparent).
+
+-  `lighting`: Adds 3D lighting effects, making the mesh appear more realistic.
+
+-  `intensity`: This can be used to apply a color scale to the mesh based on a data array, similar to a heatmap on the mesh surface
```
Comment (mamtawardhani) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:47]:
```suggestion

- `x`: An array of X-coordinates of the vertices in 3D space.
- `y`: An array of Y-coordinates of the vertices in 3D space.
- `z`: An array of Z-coordinates of the vertices in 3D space.
- `i`: An array that defines the first vertex of each triangle by its index in the `x`, `y`, and `z` arrays.
- `j`: An array that defines the second vertex of each triangle by its index in the `x`, `y`, and `z` arrays.
- `k`: An array that defines the third vertex of each triangle by its index in the `x`, `y`, and `z` arrays.
- `color`: The color of the mesh surface.
- `opacity`: A float value between 0 and 1 that sets the transparency of the mesh surface.
```

File: content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+k=[2, 3, 1, 3],
+name='y',
+showscale=True
+ )
+])
+
+fig.show()
+```
+
+This example creates a 3D mesh representing a simple triangular surface. The mesh is defined using three vertices forming a triangular shape, each associated with `x, y,` and `z` coordinates. The `colorscale` is applied to the surface, giving it a smooth gradient from yellow to magenta, representing changes in elevation or value along the `z-axis`.
```
Comment (mamtawardhani) [inline content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:81]:
```suggestion
This example creates a 3D mesh plot representing a triangular surface using Plotly. The mesh is defined with vertices specified by `x`, `y`, and `z` coordinates, forming triangular faces through the indices provided by `i`, `j`, and `k`. The `colorscale` parameter is used to apply a gradient from gold to magenta, representing the `intensity` values of each vertex, which are interpolated across the surface. The `colorbar_title` and `showscale` parameters control the appearance of the color bar and its display.
```

Comment (mamtawardhani) [reply content/plotly/concepts/graph-objects/terms/mesh3d/mesh3d.md:81]: Added a bit more of description based on the code

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @FelixOmollo thank you for contributing to Codecademy Docs, the entry is nicely written 😄 

I've suggested a few changes, could you review and modify them at your earliest convenience? Feel free to discuss any concerns, thank you! 😃

**Comment by FelixOmollo** [comment]
Hello Mamta, I have reviewed and committed your proposed changes. I would greatly appreciate your feedback.

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @FelixOmollo 😃 

The entry looks good for a second round of review! 🚀

**Comment by mamtawardhani** [comment]
> [CLA assistant check](https://cla-assistant.io/Codecademy/docs?pullRequest=5201) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=5201) before we can accept your contribution.**2** out of **3** committers have signed the CLA.✅ FelixOmollo✅ mamtawardhani❌ Your Name
> 
> **Your Name** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=5201) it.

Hey @Sriparno08 looks like the author has made some commits with the local device ID as well, could you please help resolve this?

@FelixOmollo could you please visit [this link](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user) and look through the process of adding your email id to your GitHub account.

**Comment by Sriparno08** [comment]
Okay @mamtawardhani, I'll take a look at it.

**Review by Sriparno08 — APPROVED** [review]
Looks good, @FelixOmollo!

**Comment by Sriparno08** [comment]
Hey @FelixOmollo, have you visited [this link](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user) and looked through the process of adding your email id to your GitHub account? If not, please do so that the entry can be merged.

Note: If there's no response from your side in a week, this PR will be closed unfortunately.

**Comment by Sriparno08** [comment]
Closing this PR since the author is inactive.

**Pull request closed without merging**